lib/charms/layer/jenkins/api.py

```diff
@@ -34,13 +34,21 @@
 class Api(object):
     """Encapsulate operations on the Jenkins master."""
 
+    @property
+    def url(self):
+        config = hookenv.config()
+        prefix = urlparse(config["public_url"]).path
+        if not prefix:
+            prefix = "/"
+        return "http://localhost:8080{}".format(prefix)
```

Comment:
There is no URL constant in service anymore, I removed it. A constant is no longer valid with the ability to change prefix and it was only used by API calls so it makes more sense to be in the api file rather than in the service file. This method is the replacement for the constant.